src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/aggregators/BucketizedSlidingWindow.java

```diff
@@ -41,6 +62,11 @@ public void next(SlidingWindowData e) {
         firstElement.value += e.getValue();
         add(e);
         pruneExpiredEntries(e.getTimeStamp());
+        try {
+          write(); // Try to persist the data whenever we complete writing a bucket
+        } catch (IOException ex) {
```

Comment:
Can we add a metric here tracking persistence failure?

Reply:
Done